candle-transformers/src/models/depth_anything_v2.rs

```diff
@@ -0,0 +1,535 @@
+                    let to_cat = [x, readout];
+                    let cat = Tensor::cat(&to_cat, Minus1).unwrap();
+                    self.readout_projections[i].forward(&cat).unwrap()
+                } else {
+                    xs.get(i).unwrap()
+                };
+                let x_dims = x.dims();
+
+                let x = x
+                    .permute((0, 2, 1))
+                    .unwrap()
+                    .reshape((x_dims[0], x_dims[x_dims.len() - 1], PATCH_SIZE, PATCH_SIZE))
+                    .unwrap();
+                let x = self.projections[i].forward(&x).unwrap();
+
+                self.resize_layers[i].forward(&x).unwrap()
+            })
+            .collect::<Vec<Tensor>>();
+
+        let layer_1_rn = self.scratch.layer1_rn.forward(out.get(0).unwrap())?;
+        let layer_2_rn = self.scratch.layer2_rn.forward(out.get(1).unwrap())?;
+        let layer_3_rn = self.scratch.layer3_rn.forward(out.get(2).unwrap())?;
+        let layer_4_rn = self.scratch.layer4_rn.forward(out.get(3).unwrap())?;
+
+        let path4 = self.scratch.refine_net4.forward(&layer_4_rn)?;
+
+        let res3_out = self
+            .scratch
+            .refine_net3
+            .res_conv_unit1
+            .forward(&layer_3_rn)?;
+        let res3_out = path4.add(&res3_out)?;
+        let path3 = self.scratch.refine_net3.forward(&res3_out)?;
+
+        let res2_out = self
+            .scratch
+            .refine_net2
+            .res_conv_unit1
+            .forward(&layer_2_rn)?;
+        let res2_out = path3.add(&res2_out)?;
+        let path2 = self.scratch.refine_net2.forward(&res2_out)?;
+
+        let res1_out = self
+            .scratch
+            .refine_net1
+            .res_conv_unit1
+            .forward(&layer_1_rn)?;
+        let res1_out = path2.add(&res1_out)?;
+        let path1 = self.scratch.refine_net1.forward(&res1_out)?;
+
+        let out = self.scratch.output_conv1.forward(&path1)?;
+
+        let out = out.interpolate2d(DINO_IMG_SIZE, DINO_IMG_SIZE)?;
+
+        self.scratch.output_conv2.forward(&out)
+    }
+}
+
+pub struct DepthAnythingV2<'a> {
+    pretrained: &'a DinoVisionTransformer,
+    depth_head: DPTHead,
+}
+
+impl<'a> DepthAnythingV2<'a> {
+    pub fn new(
+        pretrained: &'a DinoVisionTransformer,
+        in_channel_size: usize,
+        out_channel_sizes: Vec<usize>,
+        num_features: usize,
+        var_builder: VarBuilder,
+    ) -> Result<Self> {
+        let depth_head = DPTHead::new(
+            out_channel_sizes,
+            in_channel_size,
+            num_features,
+            false,
+            false,
+            var_builder.push_prefix("depth_head"),
+        )?;
+
+        Ok(Self {
+            pretrained,
+            depth_head,
+        })
+    }
+}
+
+impl<'a> Module for DepthAnythingV2<'a> {
+    fn forward(&self, xs: &Tensor) -> Result<Tensor> {
+        let dims = xs.shape().dims();
+        println!("DepthAnythingV2 got dims: {:?}", dims);
```

Comment:
We should not have any print statement in the model code as it can be used as a library, we also prefer not to have any form of logging. Better if you can do this on the `candle-examples` side.